scripts/stackstorm.js

```diff
@@ -397,6 +399,9 @@ module.exports = function(robot) {
       source.onerror = function (err) {
         // TODO: squeeze a little bit more info out of evensource.js
         robot.logger.error('Stream error:', err);
+        if (err.status === 401) {
+          throw err;
```

Comment:
@nmaludy I can confirm we can workaround the https://github.com/StackStorm/hubot-stackstorm/issues/157 with this logic for stream.

It's not a proper fix, since the token should be re-requested and stream re-connected properly. For some reason that doesn't happen in reality.
I would love to see the proper fix, but in a meantime we can just let the bot exit in such situations.

Reply:
that's fine, i was hoping we might have hit two birds with one stone here

i think exiting on failure is a good option to at least signal to the user / sysadmin that something is indeed broken

Reply:
@armab Is there a reason you haven't implemented it [this way](https://github.com/StackStorm/hubot-stackstorm/pull/168#pullrequestreview-248249474)?

Not saying you're doing anything wrong, I'm just curious why you wanted it implemented that way before but not now.

Reply:
@blag The line we're referring here is for `st2stream` which only includes `status` in response. There is additional logic block for `api` which implemented _that_ way:
```
if (err.status === 401 || err.message.includes('Unauthorized')) {
```

It's based on the following:
```
ERROR Stream error: Event { type: 'error', status: 401 }
ERROR {"name":"APIError","status":401,"message":"Unauthorized - Token has expired."}
```

Am I understanding your question correctly or you meant something else?

Reply:
Yep, you understood my question and answered it perfectly, thank you.